### ISSUE TITLE
Install Script + cron on Synology NAS

### ISSUE THREAD

**Issue by TonioBDS** [Taxel/PlexTraktSync#62]
Hello to the community.
My Plex Media Server is running on an unrooted Nvidia Shield Pro (2019). I've tired for a long time to install the trakt "Plex-Trakt-Scrobbler" plugin on the PMS without success.
This script seems a very good alternative to sync Plex and Trakt.

I would like to run the PlexTraktSync script on my Synology NAS. 
Then I connected to my NAS in SSH, switch to Super User... At this stage, it is important to let you know that I am quite unfamiliar with SSH commands, Python... Such that all the commands I entered, well I don't really understand the meaning of that...
Any way I did :
`cd ~
git clone https://github.com/Taxel/PlexTraktSync.git
cd PlexTraktSync
pip3 install -r requirements.txt
cp .env_example .env
python3 main.py`

However when I entered `python3 main.py`, I got the following error:
`root@Tonio-NAS:/volume1/GitProjects/PlexTraktSync# python3 main.py Traceback (most recent call last): File "main.py", line 11, in <module> from dotenv import load_dotenv File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dotenv/__init__.py", line 2, in <module> from .main import load_dotenv, get_key, set_key, unset_key, find_dotenv, dotenv_values File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dotenv/main.py", line 10, in <module> from typing import (Dict, Iterator, List, Match, Optional, # noqa ImportError: cannot import name 'TYPE_CHECKING'`

From what I understand based on my web research, I has to do with the Python version available on my NAS. I tired to install all the python package I could find on the synology and community app store but no luck.

Did anyone succeed to install this script on a Synology NAS.
The next question will be on the way to set up a repetitive scheduled task to run automatically the script every day.

Many thanks to the community and to the script developper(s).

**Comment by Taxel** [comment]
The synology NAS can run docker containers, correct?
In that case you might want to wait for a release I am planning, where I am currently building a docker container that runs this script on a cronjob automatically. It will probably take me a couple more weeks to continue, though...

**Comment by twolaw** [comment]
For your information, I run PlexTraktSync inside a docker image, it was the only solution for me on a system with python 2 only.
Here is the docker file I created : https://github.com/twolaw/PlexTraktSync/tree/docker
It's working good for me for many months now.

**Comment by twolaw** [comment]
@TonioBDS On your Synology NAS, maybe you can install python from [another repository](https://synocommunity.com/#easy-install). To have a more recent version.

[Python 3.7.7](https://synocommunity.com/package/python3) is available on SynoCommunity.

**Comment by TonioBDS** [comment]
> @TonioBDS On your Synology NAS, maybe you can install python from [another repository](https://synocommunity.com/#easy-install). To have a more recent version.
> 
> [Python 3.7.7](https://synocommunity.com/package/python3) is available on SynoCommunity.

Thankyou for your help, Actually I did exactly what you said, I even just update this Python3 apk just now and retried the command. I get always the same message:
`

admin@Tonio-NAS:/volume1/GitProjects/PlexTraktSync$ python3 main.py
Traceback (most recent call last):
  File "main.py", line 11, in <module>
    from dotenv import load_dotenv
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dotenv/__init__.py", line 2, in <module>
    from .main import load_dotenv, get_key, set_key, unset_key, find_dotenv, dotenv_values
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/dotenv/main.py", line 10, in <module>
    from typing import (Dict, Iterator, List, Match, Optional,  # noqa
ImportError: cannot import name 'TYPE_CHECKING'
admin@Tonio-NAS:/volume1/GitProjects/PlexTraktSync$

`

**Comment by TonioBDS** [comment]
It is like if the default Python3 was v3.5 ? 
But honnestly I am not comfortable with Python usage

**Comment by TonioBDS** [comment]
> The synology NAS can run docker containers, correct?
> In that case you might want to wait for a release I am planning, where I am currently building a docker container that runs this script on a cronjob automatically. It will probably take me a couple more weeks to complete it, though...

I can definitly wait a few weeks to get the script working without trouble.

Indeed I looked to the solution from @twolaw who apparently already provided a docker version of the script.
As I am not the best in command line I was wondering if script/docker installation would come through the Docker synology app or through command line.

Also I worry in advance about the Plex and trakt credentials setting in command line...
In any case a big thank you to you two

**Comment by twolaw** [comment]
Maybe you need to disable python 3.5.1 to let python 3.7 run.
I can't help you more about Synology.

**Comment by TonioBDS** [comment]
> Maybe you need to disable python 3.5.1 to let python 3.7 run. Or reboot the NAS.
> I can't help you more about Synology.

Just tried what you suggested thinking that python3 command was linked to 3.5.1 and not 3.7 : So I uninstalled Python3.5.1 and 3.7, then reinstalled using manual install and the apk corresponding to my NAS archotecture (apollolake x86-64).

But the python3 command is unknown when trying to run the 3.7 when the command python3 was recognized with the 3.5.1 version installed from the app library and edited by Synology Inc.

I suspect the Python 3.7 package installation to not be done properly...
Thank you anyway

**Comment by twolaw** [comment]
You should use the repository, not the manual installation.
https://synocommunity.com/#easy-install

**Comment by TonioBDS** [comment]
> You should use the repository, not the manual installation.
> https://synocommunity.com/#easy-install

@twolaw, Tahnk you agin for your help. Unfortunately I already tried the Python3 package from SynoCommunity repository. And there is no 3.7 version. It is only 3.5.1.

**Comment by TonioBDS** [comment]
> The synology NAS can run docker containers, correct?
> In that case you might want to wait for a release I am planning, where I am currently building a docker container that runs this script on a cronjob automatically. It will probably take me a couple more weeks to complete it, though...

No pressure of course but just to know, was there any progress on the docker version ? I still could not manage to make it work on my system... Thanks

**Comment by twolaw** [comment]
@TonioBDS 
Meanwhile, you can find docker images here : https://hub.docker.com/search?q=plextraktsync&type=image

**Comment by TonioBDS** [comment]
@twolaw, thank you for your reply, Yes I saw that there were already different docker images for plextraktsync.

As posted before I have tried to install your image but not using command line way (because I'm not good at it) but using the Docker application, donwloading the image, installing it and launching it. However, as I'm sure there are configuration that I miss doing that because I could never get to the configuration part. Then I removed it.
@twolaw, I should probably spend more time trying the commande line or search for a step by step tutorial for newby. @Taxel maybe your docker will come with a tutorial to install it via the Docker application of Synology NAS DSM... 
Crossing fingers...

Thank you guys anyway

**Comment by TonioBDS** [comment]
Well I finally put my fingers into linux... I add several python3 installed and.... enventually I managed to install Python 3.7.7.
And after redownload the package PlexTracktSync, installed the ressources.... I finally entered into the configuration (plex user name, password...) Everythings went well until the end of configuration. But when the sync actually started I got Traceback errors:

`Your user code is: 19092CAB, please navigate to https://trakt.tv/activate to authenticate
You've been successfully authenticated. With access_token f82309b8e6XXXXXXXXXXXXXXXXXX52e98a659cfd24445c588 and refresh_token 791a7be202ee7d343XXXXXXXXXXXXXXXXXXXXXXXb54b11da3118cf2cef
You are now logged into Trakt. Your Trakt credentials have been added in .env and .pytrakt.json files.
You can enjoy sync!
Check config.json to adjust settings.
If you want to change Plex or Trakt account, just edit or remove .env and .pytrakt.json files.
Starting sync Plex TonioBDS and Trakt TonioBDS
Traceback (most recent call last):
  File "main.py", line 354, in <module>
    main()
  File "main.py", line 289, in main
    trakt_watched_shows = pytrakt_extensions.allwatched()
  File "/opt/lib/python3.8/site-packages/trakt/core.py", line 539, in inner
    json_data = self._handle_request('get', url)
  File "/opt/lib/python3.8/site-packages/trakt/core.py", line 519, in _handle_request
    json_data = json.loads(response.content.decode('UTF-8', 'ignore'))
  File "/opt/lib/python3.8/json/__init__.py", line 357, in loads
  File "/opt/lib/python3.8/json/decoder.py", line 337, in decode
  File "/opt/lib/python3.8/json/decoder.py", line 355, in raw_decode
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
root@Tonio-NAS:/volume1/GitProjects/PlexTraktSync#
`
Do you know something about it ?
Many thanks !

**Comment by TonioBDS** [comment]
Wooops, too fast on my last comment and question:
I went to see the .env file
And PLEX_BASEURL is set with a strange url: https://XX-XX-XX-XX.ffaablablablablablablablablabla74d.plex.direct:28350
With XX-XX-XX-XX being set my public IP address from internet and ffaablablablablablablablablabla74d being a code that I don't know but that I prefered alter to post on the forum.

I entered this url on my web explorer and Plex asked my for my permission that ffaablablablablablablablablabla74d access to my account... Then plex.direct web page shows my media like plex.tv

After that I relaunch a python3 main.py command and.... Yahoooo ! it is syncing as I write !!

Many thanks to all of you.
Now I start looking at the cronjob !

**Comment by TonioBDS** [comment]
Btw I never took time to come back here to provide a feedback.
I've finally never used the url xxxxxx.plex.direct, replacing it with my public IP address was working but was pointing on ma backup server.
So I replaced the URL by my local IP 192.168.x.x and it worked like a charm.

Then I have set a recurring task using the task scheduler so that it run every 2 hours.
And it works !

Thank you again to all contributors ! It took me more than 6 months in total to get Trakts synchronized back to my plex!
Issue closed!

**Comment by TonioBDS** [comment]
Btw I never took time to come back here to provide a feedback.
I've finally never used the url xxxxxx.plex.direct, replacing it with my public IP address was working but was pointing on ma backup server.
So I replaced the URL by my local IP 192.168.x.x and it worked like a charm.

Then I have set a recurring task using the task scheduler so that it run every 2 hours.
And it works !

Thank you again to all contributors ! It took me more than 6 months in total to get Trakts synchronized back to my plex!
Issue closed!
